apitest.go

```diff
@@ -724,6 +724,9 @@ func (a *APITest) assertMocks() {
 		if mock.isUsed == false && mock.timesSet {
 			a.verifier.Fail(a.t, "mock was not invoked expected times")
 		}
+		if mock.numberResponses != mock.numberCloses {
+			a.verifier.Fail(a.t, fmt.Sprintf("Response body was not closed for %s", mock.request.url))
```

Comment:
This is cool. If we include this we will break existing tests for users, whilst that might be a good thing I am still hesitant to add it since it's a breaking change.

How about we log a warning to the console instead and make it the default when we go to v2?

Thoughts @enrico5b1b4, @fergusstrange, @webbgeorge ?

Reply:
Aye make it a warn for now

Reply:
this seems like a really useful feature! I also agree for the warning for now 👍

Reply:
or maybe add warning and a .CheckResponseClose() function to turn on the testing. 
is there a way we might pass in some config for testing options ....... maybe `ApiTest().Config( config ApiTestConfig)`
where the defaults keep it like it was but the hjas the option `config.FailIfResponseNotClosed = true` cause it to fail ?

Reply:
and make that the default in v2 ?

Reply:
Hey Kevin. Yes we discussed that quite recently and decided not to add a config struct for now, just to keep things as simple as possible.

Reply:
ok ... ill move a that to warn so it can get merged

Reply:
Ill make it warn so it can get merged

Reply:
@KevinJCross I tried this out on gol-services repo but it's not working for me. I haven't investigated yet, but I removed `defer res.Body.Close()` from the code and I don't see any warnings printed. Can you replicate this issue?